### PR TITLE
E2E: Migrate the `Jetpack: Plugins - Install/Remove` spec to Playwright.

### PR DESCRIPTION
**Pull request by worldomonation**
#### Changes proposed in this Pull Request

This PR migrates the aforementioned suite to run with Playwright.

Key changes:
- restructure some methods in PluginsPage.
- add or update JSDoc where appropriate.
- migrate the Selenium spec to Playwright.

#### Testing instructions

Ensure the following:
  - [x] Jetpack E2E (desktop)
  - [x] Jetpack E2E (mobile)

Related to https://github.com/Automattic/wp-calypso/issues/61936.

### CODE REVIEW

File: test/e2e/specs/jetpack/plugins__install-remove.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	} );
+
+	describe( 'Plugin: Install', function () {
+		beforeAll( async function () {
+			// Ensure known good state by removing the plugin
+			// if already installed.
+			await pluginsPage.visitPage( 'hello-dolly', siteURL );
+			if ( await pluginsPage.pluginIsInstalled() ) {
+				await pluginsPage.clickRemovePlugin();
+			}
```
Comment (worldomonation) [inline test/e2e/specs/jetpack/plugins__install-remove.ts:46]: In the past we've shied away from writing tests that alter the state of the test account due to concerns when said tests are run in parallel. 

However, I believe this guard is sufficient for this spec as it is only ever run as part of the `@jetpack` test group, which itself is not scheduled regularly (at this time). Even when scheduled, I do not anticipate this spec to encounter a situation where parallel runs are taking place.

File: packages/calypso-e2e/src/lib/pages/plugins-page.ts [hunk 2]
```diff
@@ -142,4 +171,49 @@ export class PluginsPage {
...
+	 * navigating to the `/plugins/<name>/site` endpoint, then
+	 * counting for the instances of `removeButton`.
+	 *
+	 * If the `removeButton` count is greater than 0, then the plugin
+	 * is deemed to be active on a site and thus this method returns
+	 * true. False otherwise.
+	 *
+	 * @returns {Promise<boolean>} True if plugin is active on any site. False otherwise.
+	 */
+	async pluginIsInstalled(): Promise< boolean > {
```
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/pages/plugins-page.ts:190]: I suppose this can be private?

**Review by adimoldovan — APPROVED** [review]
Looks good to me! 👍

File: packages/calypso-e2e/src/lib/pages/plugins-page.ts [hunk 3]
```diff
@@ -127,13 +154,27 @@ export class PluginsPage {
...
 	async search( query: string ): Promise< void > {
-		await this.page.click( selectors.searchIcon );
+		// On mobile viewports the Loupe icon must be
+		// clicked to activate the search field.
+		const searchInputIconLocator = this.page.locator( selectors.searchIcon );
+		await searchInputIconLocator.click();
+
 		await this.page.fill( selectors.searchInput, query );
 		await this.page.press( selectors.searchInput, 'Enter' );
-		await this.page.waitForSelector( selectors.searchResultTitle( query ) );
```
Comment (dpasque) [inline packages/calypso-e2e/src/lib/pages/plugins-page.ts:136]: I think it's fine to take this out, but i'm also okay with this one remaining here. Searching is one of those async things that would be nice to have something reliable to key off of knowing we're done, and this seems like a good, safe way to do that!

File: packages/calypso-e2e/src/lib/pages/plugins-page.ts [hunk 4]
```diff
@@ -142,4 +183,85 @@ export class PluginsPage {
...
+	/**
+	 * Toggles the plugin attribute.
+	 *
+	 * @param {PluginAttributes} target Target attribute to toggle.
+	 * @param {PluginState} state Desired end state of the attribute.
+	 */
+	async togglePluginAttribute( target: PluginAttributes, state: PluginState ): Promise< void > {
+		const toggleLocator = this.page.locator( selectors.pluginToggle( target ) );
+
+		const currentState = await this.isToggled( toggleLocator );
```
Comment (dpasque) [inline packages/calypso-e2e/src/lib/pages/plugins-page.ts]: tiny optional nit, but would you be down to name this variable something like `isToggled`? I think that makes it read more easily in the logical statement that follows!

File: test/e2e/specs/jetpack/plugins__install-remove.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+	beforeAll( async function () {
+		page = await browser.newPage();
+
+		const testAccount = new TestAccount( 'jetpackUser' );
+		await testAccount.authenticate( page );
+		// This is bit of a hacky workaround to avoid committing
+		// in plain text the URL of our test sites.
+		// @TODO We really ought to make a function in DataHelper
+		// to somehow extract the user's site.
```
Comment (dpasque) [inline test/e2e/specs/jetpack/plugins__install-remove.ts:33]: Agreed - we can also just be a bit more strict about what's in the config file. We have to clean that up soon anyway and can make sure to add it there!

**Review by dpasque — APPROVED** [review]
Nice work!! Thanks for doing this! 🚀

**Pull request merged**